pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -535,3 +541,127 @@ func (os *OpenStack) Routes() (cloudprovider.Routes, bool) {
 
 	return r, true
 }
+
+// Implementation of sort interface for blockstorage version probing
+type APIVersionsByID []apiversions_v1.APIVersion
+
+func (apiVersions APIVersionsByID) Len() int {
+	return len(apiVersions)
+}
+
+func (apiVersions APIVersionsByID) Swap(i, j int) {
+	apiVersions[i], apiVersions[j] = apiVersions[j], apiVersions[i]
+}
+
+func (apiVersions APIVersionsByID) Less(i, j int) bool {
+	return apiVersions[i].ID > apiVersions[j].ID
+}
+
+func autoVersionSelector(apiVersion *apiversions_v1.APIVersion) string {
+	switch apiVersion.ID {
+	case "v2.0":
+		return "v2"
+	case "v1.0":
+		return "v1"
+	default:
+		return ""
+	}
+}
+
+func doBsApiVersionAutodetect(availableApiVersions []apiversions_v1.APIVersion, AllowAutoSupported bool) string {
```

Comment:
Instead of duplicating the `for` loop logic, why not just do something like this: https://github.com/kubernetes/kubernetes/pull/40423#discussion_r106489621 ?